docs/proposals/last-operation-description.md

```diff
@@ -0,0 +1,97 @@
+- No guarantee that last_operation shows every operation.
+- The final_operation is the last_operation gathered before the APB is deleted.
+
+
+## Terms
+
+**Last Operation:**  the most recent operation an apb performed 
+
+**Final operation:** the operation that indicates the apb is 100% complete
+                     
+
+## Proposed Solution
+
+Using a new apb module, allow for a description to be added by the apb developer for the last operation the apb took along with the final operation the apb took. This module would take advantage of env vars,  provided to it via the downward api, that reference the pod name and namespace the apb is executing within. 
+A PR is already in place to expose this information: https://github.com/openshift/ansible-service-broker/pull/546
+When called this apb module would update known annotations on the pod ie: ```apb_last_operation``` and ``` apb_final_operation```
+with the description provided by the apb developer. This would be part of the 
+[ansible playbook modules](https://github.com/ansibleplaybookbundle/ansible-asb-modules).
+
+In order to collect this information we would use a watch via the Kubernetes client on the pod resource within the temporary namespace [Pod Rest API](https://docs.openshift.com/container-platform/3.5/rest_api/kubernetes_v1.html#list-or-watch-objects-of-kind-pod).
+This would allow us to react to changes made (i.e to the annotations) on Pod Object. Whenever a change occurred, an update to the JobStatus would happen. If the ```apb_final_operation``` annotation is present this would take precedence over the last_operation annotation. 
+Once the pod was deleted we would stop the watch on the pod and update the JobStatus ```final_operation``` annotation value.
+
+Sudo code example
+
+```go
+
+wi, err := k8client.CoreV1().Pods(ns).Watch(meta_v1.ListOptions{})
+changes := wi.ResultChan()
+    for ch := range changes{
+		if ch.Type == watch.Modified{
+			...
+		}
+		if ch.Type == watch.Deleted{
+			close(ch)
+			...
+		}
+	}
+
+```
+
+As this would block, it would need to be done in a background go routine. Using a watch in a background routine should allow us to update the JobState independent of the actual execution of the apb.    
+
+
+### Broker changes
+Below are a set of changes that I believe are in line with the current design. The exact implementation would likely differ but the gist would be the same.
+
+- A new field would be added to the JobState type:
+```Description string ``` 
+The string value , added by the apb module and gathered from the pod annotation would be stored here.
+
+- Modify the existing provision and deprovision subscribers along with the corresponding work messages.
+A namespace field would be added to these messages as the pod name is already present. 
+Inside the subscriber, the go routine to watch the pod and update the JobStatus would start when a new work msg was recieved.
+This routine would stop once the a deleted change was received in the watch.
```

Comment:
nit: This routine would stop once ~~the~~ a deleted change was received in the watch.